[ADD] general/payment_acquirers: add doc on Stripe Webhook Setup (#1335)
odoo/documentation: content/applications/general/payment_acquirers/stripe.rst

Odoo needs your **API Credentials** to connect with your Stripe account, which comprise:

- :ref:`Publishable Key <stripe/api_keys>`: The key solely used to identify the account with Stripe.
- :ref:`Secret Key <stripe/api_keys>`: The key to sign the merchant account with Stripe.

> - :ref:`Webhook Signing Secret <stripe/webhook>`: When you enabled your webhook on your Stripe

Reviewer: ```suggestion
- :ref:`Webhook Signing Secret <stripe/webhook>`: When you enable your webhook on your Stripe
```

  account, this signing secret must be set to authenticate the messages sent from Stripe to Odoo.

.. important::
   If you are trying Stripe as a test, in the **test mode**, change the **State** to *Test
   Mode*. We recommend doing this on a test Odoo database, rather than on your main database.

.. _stripe/api_keys:

Publishable and Secret keys
~~~~~~~~~~~~~~~~~~~~~~~~~~~

To retrieve the publishable and secret keys, follow this `link to your api keys
<https://dashboard.stripe.com/account/apikeys>`_, or log into your Stripe dashboard and go to
:menuselection:`Developers --> API Keys --> Standard Keys`.

.. _stripe/webhook:

Webhook Signing Secret
~~~~~~~~~~~~~~~~~~~~~~

> To retrieve the webhook signing secret, you'll first need to create this webhook. To do so, follow

Reviewer: ```suggestion
To retrieve the webhook signing secret, you first need to create a webhook. To do so, follow
```

Reviewer 2: Indeed.

The present simple is the correct grammatical tense for general truths, things that always happen. The future is only for events that happen in the future, relative to the context. (for example here "to do this, you first need to create a webhook. After two weeks you will have to create a new one.")

See https://www.odoo.com/documentation/14.0/contributing/documentation/content_guidelines.html#grammatical-tenses

this `link to your webhooks <https://dashboard.stripe.com/webhooks>`_, or log into your Stripe
dashboard and go to :menuselection:`Developers --> Webhooks`.

Then, click on **Add endpoint** in your **Hosted endpoints**. A form opens, where you'll need to add

> two informations:

Reviewer: ```suggestion
two information:
```

Reviewer 2: This suggestion isn't correct.

"information" is uncountable. You would have "two pieces of information" for example.
But "information" isn't the correct word here anyway. It would rather be something such as "data" (you enter data in fields)

- In the **Endpoint URL**, enter ``https://yourcompany.odoo.com/payment/stripe/webhook`` after
  replacing *yourcompany.odoo.com* by your address.

> - At the end of the form, you can **Select events** to listen to. Click on it, and search for
>   **checkout.session.completed**. Add it. It is possible to select other events but they are
>   currently not processed by Odoo.

Reviewer: ```suggestion
- At the end of the form, you can **Select events** to listen to. Click on it and, in the **Checkout** section, select
 **checkout.session.completed**.
 
 .. note::
    It is possible to select other events but they are currently not processed by Odoo.
```

Odoo needs your **API Credentials** to connect with your Stripe account, which comprise:

- :ref:`Publishable Key <stripe/api_keys>`: The key solely used to identify the account with Stripe.
- :ref:`Secret Key <stripe/api_keys>`: The key to sign the merchant account with Stripe.
- :ref:`Webhook Signing Secret <stripe/webhook>`: When you enable your webhook on your Stripe
  account, this signing secret must be set to authenticate the messages sent from Stripe to Odoo.

.. important::
   If you are trying Stripe as a test, in the **test mode**, change the **State** to *Test
   Mode*. We recommend doing this on a test Odoo database, rather than on your main database.

.. _stripe/api_keys:

Publishable and Secret keys
~~~~~~~~~~~~~~~~~~~~~~~~~~~

> To retrieve the publishable and secret keys, follow this `link to your api keys

Reviewer 2: ```suggestion
To retrieve the publishable and secret keys, follow this `link to your API keys
```
API is in capital letters. (acronym)

<https://dashboard.stripe.com/account/apikeys>`_, or log into your Stripe dashboard and go to
:menuselection:`Developers --> API Keys --> Standard Keys`.

.. _stripe/webhook:

Webhook Signing Secret
~~~~~~~~~~~~~~~~~~~~~~

> To retrieve the webhook signing secret, you first need to create a webhook. To do so, follow
> this `link to your webhooks <https://dashboard.stripe.com/webhooks>`_, or log into your Stripe
> dashboard and go to :menuselection:`Developers --> Webhooks`.
> 
> Then, click on **Add endpoint** in your **Hosted endpoints**. A form opens, where you'll need to add
> two data:

Reviewer 2: ```suggestion
To retrieve the webhook signing secret, you first need to create a webhook.

To do so, follow this `link to your Stripe webhooks <https://dashboard.stripe.com/webhooks>`_, or
log into your Stripe dashboard and go to :menuselection:`Developers --> Webhooks`. Then, click on
*Add endpoint* in your **Hosted endpoints** and add the following data:
```

- I would separate the ideas differently. The first paragraph explains what we need to do, then the second paragraph explains how to do it
- adding "Stripe" in the link text makes the link more accurate. It also helps the referencing for search engines
- as explained previously, don't use a future tense for general truths and immediate actions. Also, "data" is uncountable, as was "information" before. Better use "add the following data".

> - In the **Endpoint URL**, enter ``https://yourcompany.odoo.com/payment/stripe/webhook`` after
>   replacing *yourcompany.odoo.com* by your address.

Reviewer 2: ```suggestion
- | In the **Endpoint URL**, enter your Odoo database's URL followed by ``/payment/stripe/webhook``.
  | For example: ``https://yourcompany.odoo.com/payment/stripe/webhook``
```

Here is a suggestion to make it sure that some users won't just copy-paste the example URL.

- At the end of the form, you can **Select events** to listen to. Click on it and, in the
  **Checkout** section, select **checkout.session.completed**.

> .. note::
>   It is possible to select other events but they are currently not processed by Odoo.

Reviewer 2: ```suggestion
.. note::
   It is possible to select other events, but they are currently not processed by Odoo.
```
- [be consistent with the indentation ](https://www.odoo.com/documentation/14.0/contributing/documentation/rst_guidelines.html#be-consistent-with-indentation)
- add a comma before "but"

> When you click on **Add endpoint**, your Webhook will be configured. You can then click on
> **reveal** to see your signing secret.

Reviewer 2: ```suggestion
When you click on **Add endpoint**, your Webhook is configured. You can then click on **reveal** to
display your signing secret.
```
- present tense
- see --> display
- break the ligne only after the 100th character. see  [RST guidelines](https://www.odoo.com/documentation/14.0/contributing/documentation/rst_guidelines.html#start-a-new-line-before-the-100th-character)
